### PR TITLE
Add explicit check to silence warning on MacOS

### PR DESCRIPTION
**Pull request by remilapeyre**
**skip news**
**skip bpo**

### CODE REVIEW

File: Modules/posixmodule.c [hunk 1]
```diff
@@ -1261,7 +1261,7 @@ _Py_Sigset_Converter(PyObject *obj, void *addr)
     long signum;
     int overflow;
 
-    if (sigemptyset(mask)) {
+    if (sigemptyset(mask) < (0)) {
```
Comment (matrixise) [inline Modules/posixmodule.c:1265]: why the parentheses?

Comment (remilapeyre) [reply Modules/posixmodule.c:1265]: It's the way to disable it with clang:
```
./Modules/posixmodule.c:1266:9: warning: code will never be executed [-Wunreachable-code]
        PyErr_SetFromErrno(PyExc_OSError);
        ^~~~~~~~~~~~~~~~~~
./Modules/posixmodule.c:1264:29: note: silence by adding parentheses to mark code as explicitly dead
    if (sigemptyset(mask) < 0) {
                            ^
                            /* DISABLES CODE */ ( )
```

Comment (remilapeyre) [reply Modules/posixmodule.c:1265]: Maybe I should put a comment about it?

**Comment by methane** [comment]
I don't see such warning on macOS.

**Comment by remilapeyre** [comment]
Hi @methane, I am using clang version 10.0.0 and I only get the warning when I am not using `--with-pydebug`. I still get it on master (cc60cdd9c44dd15e441603ee5f78e09ea3e76929):

```
➜  cpython git:(cc60cdd9c4) make distclean -s 2>/dev/null && ./configure --with-openssl=$(brew --prefix openssl) 2>&1 >/dev/null && make clean -s 2>/dev/null&& make -sj
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/c++/4.2.1
./Modules/posixmodule.c:1266:9: warning: code will never be executed [-Wunreachable-code]
        PyErr_SetFromErrno(PyExc_OSError);
        ^~~~~~~~~~~~~~~~~~
1 warning generated.
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libpython3.8m.a(dynamic_annotations.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libpython3.8m.a(pymath.o) has no symbols
 CC='gcc' LDSHARED='gcc -bundle -undefined dynamic_lookup -L/usr/local/opt/zlib/lib -L/usr/local/opt/zlib/lib  ' OPT='-DNDEBUG -g -fwrapv -O3 -Wall' 	_TCLTK_INCLUDES='' _TCLTK_LIBS='' 	./python.exe -E ./setup.py -q build
ld: warning: directory not found for option '-L/usr/lib/termcap'
clang: warning: -framework Tk: 'linker' input unused [-Wunused-command-line-argument]
clang: warning: -framework Tk: 'linker' input unused [-Wunused-command-line-argument]
ld: warning: could not create compact unwind for _ffi_call_unix64: does not use RBP or RSP based frame

Python build finished successfully!
The necessary bits to build these optional modules were not found:
_dbm                  _sqlite3              _uuid
nis                   ossaudiodev           spwd
To find the necessary bits, look in setup.py in detect_modules() for the module's name.


The following modules found by detect_modules() in setup.py, have been
built by the Makefile instead, as configured by the Setup files:
_abc                  atexit                pwd
time

```

File: Modules/posixmodule.c (hunk 1 above)
Comment (methane) [reply Modules/posixmodule.c:1265]: Yes, please.

Comment (remilapeyre) [reply Modules/posixmodule.c:1265]: Thanks @methane, I added a comment about it

**Pull request merged**